### PR TITLE
Update PR template to include commit message instructions

### PR DESCRIPTION
**Pull request by chenrui333**
Propose to update the PR template to make the commit message instruction more accessible.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
+
+Please make sure the commit message would be as follows:
+# For new formula
+The commit message should be `<formula> <version> (new formula)`
+
+# For bump the version of the formula (also highly recommend to use `brew bump-formula-pr <formula>` command)
+The commit message should be `<formula> <version>`
+
+# For submitting a new patch for the formula
+The commit message should be `<formula>: short-description-for-the-change`
```
Comment (EricFromCanada) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
The commit message should be `<formula>: short-description-of-the-change`
```

**Review by EricFromCanada — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,3 +1,17 @@
+<!--
+
+Please make sure the commit message would be as follows:
+# For new formula
+The commit message should be `<formula> <version> (new formula)`
+
+# For bump the version of the formula (also highly recommend to use `brew bump-formula-pr <formula>` command)
```
[the remark this answers is not in the document]
Comment (issyl0) [reply .github/PULL_REQUEST_TEMPLATE.md]: Another suggestion:

```suggestion
# Using `brew bump-formula-pr` to bump a version of a formula is highly recommended as it does everything for you. If that command doesn't work or you choose to do the version bump manually, your commit message should look like:
```

**Comment by MikeMcQuaid** [comment]
I like the intention here but it does make the PR template a _lot_ longer. I think this might be a better fit for a linked document e.g. `CONTRIBUTING.md`?

**Comment by chenrui333** [comment]
@MikeMcQuaid, not everyone likes to read the doc (myself included for some cases too :) ), so the intention is just to experiment to see if we can formalize the commit messages.

**Comment by chenrui333** [comment]
Thanks @EricFromCanada @issyl0 for the feedback, PR template updated.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]
> @MikeMcQuaid, not everyone likes to read the doc (myself included for some cases too :) ), so the intention is just to experiment to see if we can formalize the commit messages.

I don't think making this document longer is likely to do that. This ends up deprioritising the existing checklist. What about a dedicated document just for commit message formats which you add to the checklist with a link?

**Comment by SMillerDev** [comment]
> not everyone likes to read the doc

What makes you think they'll read this docs in that case?

**Comment by chenrui333** [comment]
I like the audit check idea! I don't quite like documentation in general though. 😄 

Yeah, the idea of doing the PR template update rather than another contributing doc is it would be the most immediate feedback.

This actually push me to think there must be some sort of `probot` to do this kind of check (while I am typing this comment).

**Comment by chenrui333** [comment]
I will work on the doc and audit ideas this weekend.

**Comment by MikeMcQuaid** [comment]
Sorry, passing on this for now @chenrui333!

**Pull request closed without merging**